### ISSUE TITLE
AKS : Terraform wants to recreate cluster after nodepool grow from the portal

### ISSUE THREAD

**Issue by seboss666** [terraform-providers/terraform-provider-azurerm#4825]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
terraform 0.11.14
azurerm 1.32.1

### Affected Resource(s)

* `azurerm_kubernetes_cluster`

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster" "aks" {
#  name                = "AKS-${var.environment}"
  name                = "${var.aks_cluster_name}"
  location            = "${var.location}"
  resource_group_name = "${var.resource_group_name}"
  dns_prefix          = "aks-${var.environment}"
  kubernetes_version  = "${var.aks_kubernetes_version}"

  lifecycle {
    #prevent_destroy = true
    ignore_changes = [ "agent_pool_profile.0.count" ]
  }

  agent_pool_profile {
    name            = "default"
    count           = "${var.aks_node_count}"
    vm_size         = "${var.aks_node_vm_size}"
    os_type         = "Linux"
    os_disk_size_gb = "${var.aks_node_vm_disk_size}"
    max_pods        = "${var.aks_node_max_pods}"
    vnet_subnet_id  = "${azurerm_subnet.subnet_aks_nodepod.id}"
  }

  linux_profile {
    admin_username = "azureuser"

    ssh_key = {
      key_data = "${var.ssh_key}"
    }
  }

  service_principal {
    client_id     = "${azuread_application.aks_ad_app.application_id}"
    client_secret = "${azuread_service_principal_password.aks_ad_app.value}"
  }

  role_based_access_control {
    enabled = true

    azure_active_directory {
      client_app_id     = "${var.aks_ad_client_app_id}"
      server_app_id     = "${var.aks_ad_server_app_id}"
      server_app_secret = "${var.aks_ad_server_app_password}"
      tenant_id         = "${data.azurerm_client_config.current.tenant_id}"
    }

  }

  network_profile {
    service_cidr       = "${var.subnet_service_prefix}"
    dns_service_ip     = "${var.aks_dns_service_ip}"
    docker_bridge_cidr = "172.17.0.1/16"                # Default https://docs.microsoft.com/en-us/azure/aks/configure-azure-cni#plan-ip-addressing-for-your-cluster
    network_plugin     = "azure"

  }

  addon_profile {
    oms_agent {
      enabled                    = true
      log_analytics_workspace_id = "${var.log_analytics_workspace_id}"
    }

    http_application_routing {
      enabled = false
    }
  }

  depends_on = ["azuread_service_principal.aks_ad_app"]

  tags = "${merge(
                 var.custom_tags,
                 var.default_tags,
                 map("lbnref","${var.cloud_service_lbnref}"),
                )
              }"
}

```

### Debug Output

https://gist.github.com/seboss666/d6d99160100f256b53fab4abebab4175

### Expected Behavior

Node numbers changed manually via azure portal and put back to initial configuration. Terraform apply should have said nothing.

### Actual Behavior

Terraform wants to recreate the cluster, because some properties changed in the resource json description. httpapplicationrouting property was written in CamelCase and is now in full lower case in the json. There also is something in the terraform output about the load-balancer sku going from "Basic" to "basic" (also a "lowering" case forcing the new resource thing).

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Upgrading the provider (as worked on in #4429 ) did not help. I could not try terraform 0.12.x as there are other dependencies not yet ready to be migrated. In fact I did try but I got too many errors.

Trying to remove the cluster from terraform state and reimport it did not help either, the behaviour was exactly the same.
 
### References

* #0000

**Comment by tombuildsstuff** [comment]
hi @seboss666 

Thanks for opening this issue :)

Taking a look through this appears to be a duplicate of #3835 - rather than having multiple issues open tracking the same thing I'm going to close this issue in favour of that one; would you mind subscribing to #3835 for updates?

Thanks!

**Comment by seboss666** [comment]
Please no, it's more specific, the resource has been "touched" outside terraform (via azure portal), and now I can't manage it without delete the resource first :/

I would like to know if there is a way to get it back on track without deleting everything related :/

**Comment by jfcoz** [comment]
@tombuildsstuff , it's not a duplicate of https://github.com/terraform-providers/terraform-provider-azurerm/issues/3835, here there is only one agent_pool_profile, this is not the same problem.
Can you re-open this issue please ?

**Comment by tombuildsstuff** [comment]
@jfcoz @seboss666 the fix for this will be solved via the PR which resolves #3835 - which is why this is closed in favour of that issue (at this time we're currently waiting on Microsoft to be able to ship said fix)
